clientupdate/clientupdate.go

```diff
@@ -249,16 +250,20 @@ func (up *Updater) getUpdateFunction() (fn updateFunction, canAutoUpdate bool) {
 	return nil, false
 }
 
+var canAutoUpdate lazy.SyncValue[bool]
+
 // CanAutoUpdate reports whether auto-updating via the clientupdate package
 // is supported for the current os/distro.
 func CanAutoUpdate() bool {
-	if version.IsMacSysExt() {
-		// Macsys uses Sparkle for auto-updates, which doesn't have an update
-		// function in this package.
-		return true
-	}
-	_, canAutoUpdate := (&Updater{}).getUpdateFunction()
-	return canAutoUpdate
+	return canAutoUpdate.Get(func() bool {
```

Comment:
I don't think this is worth caching, the first time `onTailnetDefaultAutoUpdate` will run the value in prefs will be set and future calls will bail here: https://github.com/tailscale/tailscale/pull/15395/files#diff-314ba0d799f70c8998940903efb541e511f352b39a9eeeae8d475c921d66c2acR3471-R3475

Could you remove this caching wrapper to make the code a bit more readable?

Reply:
Not on platforms where CanAutoUpdate is false.

Then we never set Apply and this code runs on every reconnect.

Including all the calls to:

```go
func haveExecutable(name string) bool {
	path, err := exec.LookPath(name)
	return err == nil && path != ""
}
```

Which stats all over the disk.

Reply:
arch is the worst case.

It'll run this each time:

```go
		case distro.Arch:
			if up.archPackageInstalled() {
				// Arch update func just prints a message about how to update,
				// it doesn't support auto-updates.
				return up.updateArchLike, false
			}


func (up *Updater) archPackageInstalled() bool {
	err := exec.Command("pacman", "--query", "tailscale").Run()
	return err == nil
}
```

Reply:
ah, that's fair. SGTM then!